### DISCUSSION TITLE
write pictures to external flash, and read from flash

### DISCUSSION THREAD

**Discussion by epikao** [stm32duino/Arduino_Core_STM32#1632] [Q&A] [answered]
Hello, I am using PlatformIO with your Arduino Framework and disco_f746ng board. My board is very similar to the origin discovery board but has additional 16MB external flash ram (customized).
Are there examples how to enable QSPI, write pictures content to an external flash, and then read the content...

Do you have some advice for me here for the right approach?

Thank you

**Comment by ABOSTM** [comment]
Hi @epikao,
As far as stm32duino is concerned, we support officially only this QSPI library
https://github.com/stm32duino/MX25R6435F
It is dedicated to MX25R6435F flash, and there are example to read/write from the application.
I don't know how to write flash from external tools (AFAIU it is your need),
nevertheless there is an STMicroelectronics github (independent from Arduino) that may help you
https://github.com/STMicroelectronics/stm32-external-loader
Also, you can have a look at Stm32CubeProgrammer, for which user manual talk about external memories:
https://www.st.com/resource/en/user_manual/dm00403500-stm32cubeprogrammer-software-description-stmicroelectronics.pdf

If not enough, you can enquire from st community.
https://community.st.com/s/topiccatalog

**Reply by epikao** [reply]
Ok thank you for this infos.

With the STM32CubeProgrammer I could flash the external memory (bin-File, pictures).
Finally I only need to activate the QSPI peripheral and have it go into memory mapped mode.
Which files / code lines should I add into the project to activate this hardware?

Thank you

**Comment by ABOSTM** [comment]
I am not sure to understand your question, do you want to use MX25R6435F library ?
in such case you need to install the library with Arduino IDE for example.

**Comment by epikao** [comment, marked as answer]
I will try to include following files:
https://github.com/lvgl/lv_port_stm32f746_disco/blob/master/Utilities/Components/n25q128a/n25q128a.h
https://github.com/lvgl/lv_port_stm32f746_disco/blob/master/Utilities/STM32746G-Discovery/stm32746g_discovery_qspi.h
https://github.com/lvgl/lv_port_stm32f746_disco/blob/master/Utilities/STM32746G-Discovery/stm32746g_discovery_qspi.c

and then:
```
BSP_QSPI_Init();
BSP_QSPI_EnableMemoryMappedMode();
```
hope that will work x-D

**EDIT:**
Just tested, and it works. So problem solved :-D
